src/test/java/com/sprint/mission/discodeit/integration/ChannelTest.java

```diff
@@ -0,0 +1,164 @@
+package com.sprint.mission.discodeit.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sprint.mission.discodeit.dto.request.ChannelUpdateRequestDto;
+import com.sprint.mission.discodeit.dto.request.PrivateChannelCreateRequestDto;
+import com.sprint.mission.discodeit.dto.request.PublicChannelCreateRequestDto;
+import com.sprint.mission.discodeit.dto.response.ChannelDto;
+import com.sprint.mission.discodeit.entity.ChannelType;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.exception.ErrorResponse;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ActiveProfiles("test")
+@Transactional
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ChannelTest {
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private UserRepository userRepository;
+
+
+    @Test
+    @DisplayName("공개 채널 생성 -성공")
+    void testCreatePublicChannelSuccess() throws Exception {
+        PublicChannelCreateRequestDto dto = new PublicChannelCreateRequestDto("public", "public channel");
+        String json = objectMapper.writeValueAsString(dto);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> requestEntity = new HttpEntity<>(json, headers);
+        ResponseEntity<ChannelDto> response = restTemplate.postForEntity("/api/channels/public", requestEntity, ChannelDto.class);
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertNotNull(response.getBody().getId());
+        assertEquals("public", response.getBody().getName());
+        assertEquals(ChannelType.PUBLIC, response.getBody().getType());
```

Comment:
가독성을 위해 given-when-then을 띄어쓰기로라도 구분해볼까요?

```suggestion
        
        PublicChannelCreateRequestDto dto = new PublicChannelCreateRequestDto("public", "public channel");
        String json = objectMapper.writeValueAsString(dto);
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_JSON);
        HttpEntity<String> requestEntity = new HttpEntity<>(json, headers);
        
        ResponseEntity<ChannelDto> response = restTemplate.postForEntity("/api/channels/public", requestEntity, ChannelDto.class);
        
        assertEquals(HttpStatus.CREATED, response.getStatusCode());
        assertNotNull(response.getBody().getId());
        assertEquals("public", response.getBody().getName());
        assertEquals(ChannelType.PUBLIC, response.getBody().getType());
```